torch/fx/graph.py

```diff
@@ -984,13 +986,31 @@ def create_node(self, op: str, target: 'Target',
         name = self._graph_namespace.create_name(candidate, None)
         n = Node(self, name, op, target, args, kwargs, type_expr)
 
+        if self._node_metadata_hook is not None:
+            self._node_metadata_hook(n)
+
         self._graph_namespace.associate_name_with_obj(name, n)
 
         self._insert(n)
         self._find_nodes_lookup_table.insert(n)
         self._len += 1
         return n
 
+    @contextlib.contextmanager
+    def _set_node_metadata_hook(self, f):
```

Comment:
I think we should call this "_set_create_node_hook", due to the way that this hook is triggered exactly when we call "create_node" API. This way I think it's more clear what the hook is doing.